### PR TITLE
specs-go/v1: add annotations field to descriptor

### PR DESCRIPTION
**Pull request by stevvooe**
Signed-off-by: Stephen J Day <stephen.day@docker.com>

@opencontainers/image-spec-maintainers  PTAL

### CODE REVIEW

**Comment by crosbymichael** [comment]
LGTM

**Comment by wking** [comment]
9bdb5c5b looks good to me.

#501 also left them out of the schema, but #533 is fixing that missing
piece [1].

[1]: https://github.com/opencontainers/image-spec/pull/533#discussion_r98120475

**Comment by vbatts** [comment]
LGTM

**Comment by stevvooe** [comment]
cc @philips @jbouzane @brendandburns @jstarks

**Comment by vbatts** [comment]
bump

**Comment by stevvooe** [comment]
LGTM

File: specs-go/v1/descriptor.go [hunk 1]
```diff
@@ -30,4 +30,7 @@ type Descriptor struct {
 
 	// URLs specifies a list of URLs from which this object MAY be downloaded
 	URLs []string `json:"urls,omitempty"`
+
+	// Annotations contains arbitray metadata relating to the targeted content.
+	Annotations map[string]string
```
Comment (vbatts) [inline specs-go/v1/descriptor.go]: from #570, this ought to have ``json:"annotations,omitempty"``

**Comment by stevvooe** [comment]
@vbatts Added `omitempty`.

**Comment by vbatts** [comment]
LGTM

File: specs-go/v1/descriptor.go [hunk 2]
```diff
@@ -30,4 +30,7 @@ type Descriptor struct {
 
 	// URLs specifies a list of URLs from which this object MAY be downloaded
 	URLs []string `json:"urls,omitempty"`
+
+	// Annotations contains arbitray metadata relating to the targeted content.
```
Comment (jonboulle) [inline specs-go/v1/descriptor.go]: arbitrary

Comment (stevvooe) [inline specs-go/v1/descriptor.go]: Fixed.

**Comment by vbatts** [comment]
LGTM

**Comment by philips** [comment]
LGTM

**Comment by jbouzane** [comment]
LGTM

**Comment by jonboulle** [comment]
lgtm

**Pull request merged**